File: src/main/java/me/rsman/BetterMinecraftCore/configs/containers/AttributeLangContainer.java
package me.rsman.BetterMinecraftCore.configs.containers;

import me.rsman.BetterMinecraftCore.BetterMinecraftCore;
import me.rsman.BetterMinecraftCore.configs.ConfigLoader;
import me.rsman.BetterMinecraftCore.enums.EAttributes;
import me.rsman.BetterMinecraftCore.enums.EEnchants;

import java.util.HashMap;
import java.util.Map;

public class AttributeLangContainer {
    private static AttributeLangContainer instance;

    private static void setInstance(AttributeLangContainer instance) {
        AttributeLangContainer.instance = instance;
    }

    public static AttributeLangContainer getInstance() {
        return instance;
    }

    public static void load(){
        setInstance(null);
        BetterMinecraftCore.getInstance().getLogger().info("§3Loading Attributes lang...");
        AttributeLangContainer enchantLangContainerInstance = ConfigLoader.loadConfig("lang/attributes", AttributeLangContainer.class);
        BetterMinecraftCore.getInstance().getLogger().info("§bLoaded Attributes lang.");
        setInstance(enchantLangContainerInstance);
    }
    private Map<String, String> translations;

    public AttributeLangContainer() {}

    public void setTranslations(Map<String, String> translations) {
        for (Map.Entry<String, String> item: translations.entrySet()) {
            if(!EAttributes.getAllKeys().contains(item.getKey())){
                BetterMinecraftCore.getInstance().getLogger().info(item.getKey());
                translations.remove(item.getKey());
            }
        }
        this.translations = translations;
    }

    public Map<String, String> getTranslations() {
        return translations;
    }
    public String getTranslation(String key) {
        return translations.get(key);
    }
}

File: src/main/kotlin/me/rsman/BetterMinecraftCore/configs/containers/AttributeLangContainer.kt
package me.rsman.BetterMinecraftCore.configs.containers

import me.rsman.BetterMinecraftCore.enums.EAttributes
import me.rsman.BetterMinecraftCore.BetterMinecraftCore
import me.rsman.BetterMinecraftCore.configs.ConfigLoader

class AttributeLangContainer {
    var translations: Map<String?, String?>? = null
        private set

    fun setTranslations(translations: MutableMap<String?, String?>) {
        for ((key) in translations) {
            if (!EAttributes.keys.contains(key)) {
                BetterMinecraftCore.instance.logger.info(key)
                translations.remove(key)
            }
        }
        this.translations = translations
    }

    fun getTranslation(key: String?): String? {
        return translations!![key]
    }

    companion object {
        @JvmStatic
        var instance: AttributeLangContainer? = null
            private set

        fun load() {
            instance = null
            BetterMinecraftCore.instance.logger.info("§3Loading Attributes lang...")
            val enchantLangContainerInstance = ConfigLoader.loadConfig("lang/attributes", AttributeLangContainer::class.java)
            BetterMinecraftCore.instance.logger.info("§bLoaded Attributes lang.")
            instance = enchantLangContainerInstance
        }
    }
}